minCostFlow.py

```diff
@@ -97,7 +132,17 @@ def write_output_to_sif(G,out_file_name,idDict):
         if node2 in ["source","target"]:
             continue
         numE+=1
-        out_file.write(node1+"\t"+node2+"\n")
+
+        edge = (node1, node2)
+        sorted_edge = tuple(sorted(edge))
+
+        if edge in directed_dict:
+            out_file.write(node1+"\t"+node2+"\t"+"D"+"\n")
+        elif sorted_edge in undirected_dict:
+            out_file.write(node1+"\t"+node2+"\t"+"U"+"\n")
```

Comment:
Should we write out the edge in sorted_edge order instead of node1, node 2? Would that help make the method more deterministic?